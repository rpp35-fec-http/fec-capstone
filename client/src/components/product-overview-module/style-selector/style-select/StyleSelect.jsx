import React from 'react';
import Style from '../style-select-item/StyleSelectItem.jsx';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

const StyleSelect = ({ styles, handleStyleClick, currentStyle }) => {
  //todo use styles.skus for dropdown menus
  const imageListStyle = {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr 1fr 1fr',
    gridTemplateRows: 'auto',
  };
  return (
    <>
      <Typography variant='h3'>{currentStyle.name}</Typography>
      <Box style={imageListStyle}>
        {styles.results.map((style, index) => (
          <Style
            style_id={style.style_id}
            key={index}
            image={style.photos[0].thumbnail_url}
            handleStyleClick={handleStyleClick}
            currentStyle={currentStyle}
          />
        ))}
      </Box>
    </>
  );
};

export default StyleSelect;
