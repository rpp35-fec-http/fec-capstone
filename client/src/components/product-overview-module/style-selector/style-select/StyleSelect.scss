.style-list-images {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: auto;
  justify-items: center;
  gap: 5px;
}

.style-list {
  padding: 15px;
  .style-list-title {
    display: flex;
    gap: 15px;
  }
}
