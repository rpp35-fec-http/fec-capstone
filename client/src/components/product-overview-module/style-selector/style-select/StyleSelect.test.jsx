import React from 'react';
import renderer from 'react-test-renderer';
import StyleSelect from './StyleSelect.jsx';

it('renders correctly', () => {
  const styles = { style_id: 34, results: [] };
  const tree = renderer
    .create(
      <StyleSelect
        styles={styles}
        handleStyleClick={'test'}
        currentStyle={'test'}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
