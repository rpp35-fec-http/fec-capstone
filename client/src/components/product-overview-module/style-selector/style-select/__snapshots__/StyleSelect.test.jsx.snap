// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-gepadz-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:3rem;line-height:1.167;letter-spacing:0em;}",
      }
    }
    data-emotion="css gepadz-MuiTypography-root"
  />,
  <h3
    className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
  />,
  <div
    className="MuiBox-root css-0"
    style={
      Object {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr 1fr",
        "gridTemplateRows": "auto",
      }
    }
  />,
]
`;
